File: LeetCode/Medium/238-product-of-array-except-self/try-1.py
/*
 * Problem #238: Product of Array Except Self
 * Difficulty: Medium
 * Submission: Try 1
 * status: Accepted
 * Language: python
 * Date: 4/5/2026, 11:41:26 AM
 * Link: https://leetcode.com/problems/product-of-array-except-self/
 */

class Solution(object):
    def productExceptSelf(self, nums):
        res = [1] * len(nums)

        for i in range(1, len(nums)):
            res[i] = res[i-1] * nums[i-1]

        right = 1
        for i in range(len(nums)-1, -1, -1):
            res[i] *= right
            right *= nums[i]

        return res
        

File: LeetCode/Easy/628-maximum-product-of-three-numbers/try-1.java
/*
 * Problem #628: Maximum Product of Three Numbers
 * Difficulty: Easy
 * Submission: Try 1
 * status: Accepted
 * Language: java
 * Date: 7/26/2026, 7:50:08 PM
 * Link: https://leetcode.com/problems/maximum-product-of-three-numbers/
 */

import java.util.Arrays;

class Solution {
    public int maximumProduct(int[] nums) {
        Arrays.sort(nums);

        int n = nums.length;

        int product1 = nums[n - 1] * nums[n - 2] * nums[n - 3];
        int product2 = nums[0] * nums[1] * nums[n - 1];

        return Math.max(product1, product2);
    }
}
